### PR TITLE
Fixes for some stonith_admin output errors

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

**Review by kgaillot — COMMENTED** [review]
There probably was a desire to avoid a dependency on fence-agents-common at one time. I think it would be fine though, since this would be just for the pacemaker-cts package, which most users won't ever install.

Alternatively, since this is a dummy agent, validate-all doesn't have to be legit, you could just implement an action that validates a single parameter to have something to trigger a desired result.

**Comment by kgaillot** [comment]
P.S. wait a while to push an update, CI is rebuilding ...

File: lib/fencing/st_client.c [hunk 1]
```diff
@@ -1763,9 +1763,16 @@ stonith_api_validate(stonith_t *st, int call_options, const char *rsc_id,
...
-                       "Agent %s not found or does not support validation",
-                       agent);
+
+            if (error_output) {
+                *error_output = crm_strdup_printf("Agent %s not found or does not support validation",
+                                                  agent);
+            } else {
+                crm_perror(LOG_ERR,
+                           "Agent %s not found or does not support validation",
+                           agent);
```
[the remark this answers is not in the document]
Comment (clumens) [reply lib/fencing/st_client.c]: Fixed.

File: lib/fencing/st_client.c [hunk 2]
```diff
@@ -1760,19 +1760,32 @@ stonith_api_validate(stonith_t *st, int call_options, const char *rsc_id,
...
 #endif
 
+        case st_namespace_invalid:
+            errno = ENOENT;
+            rc = -errno;
+
+            if (error_output) {
+                *error_output = crm_strdup_printf("Agent %s not found", agent);
+            } else {
+                crm_perror(LOG_ERR, "Agent %s not found", agent);
```
Comment (kgaillot) [inline lib/fencing/st_client.c:1770]: missed one perror

**Review by kgaillot — COMMENTED** [review]
looks good, one small thing

File: lib/fencing/st_client.c (hunk 2 above)
Comment (clumens) [reply lib/fencing/st_client.c:1770]: Fixed.

File: lib/fencing/st_client.c [hunk 3]
```diff
@@ -1760,19 +1760,32 @@ stonith_api_validate(stonith_t *st, int call_options, const char *rsc_id,
...
 #endif
 
+        case st_namespace_invalid:
+            errno = ENOENT;
+            rc = -errno;
+
+            if (error_output) {
+                *error_output = crm_strdup_printf("Agent %s not found", agent);
+            } else {
+                crm_err(LOG_ERR, "Agent %s not found", agent);
```
Comment (kgaillot) [inline lib/fencing/st_client.c:1770]: s/LOG_ERR/

Comment (clumens) [reply lib/fencing/st_client.c]: Fixed.

**Pull request merged**